de4dot.code/deobfuscators/Dotfuscator/StringDecrypter.cs

```diff
@@ -105,7 +120,13 @@ public class StringDecrypterInfo {
 			byte key = (byte)(info.magic + value);
 			for (int i = 0; i < chars.Length; i++) {
 				char c = chars[i];
-				byte b1 = (byte)((byte)c ^ key++);
+				byte b1;
+				if (info.proVersion) {
```

Comment:
Aren't these two blocks doing the same thing, i.e., `c^key++`

Reply:
yes ,they are do the same thing